src/App/App.tsx

```diff
@@ -184,6 +184,13 @@ export const App: FunctionComponent = () => {
     });
   }
 
+  function closeAndBackToAdmin(id: string) {
+    updateState((state) => {
+      if (id === state.activeTab) setActiveTab(state.tabs[0].id);
```

Comment:
`updateState` is calling itself recursively (implicitly in `setActiveTab`). I'm not sure if this is a good idea.

Reply:
I don't think the `updateState` in line 188 is required since both actions do the state update and react batches them together.

Reply:
without it, this code was acting upon the state from moment of producing the closure, i.e. when the import tab was the active tab, and the `if` condition was always true.

Not sure about the former comment, let me dig deeper, but this is indeed smelly.

Reply:
> without it, this code was acting upon the state from moment of producing the closure

Yes, that's a common issue and hence the `useCallback(..., [deps])` and such.
But if you need a fresh copy of state, then you might want to keep `updateState` but inline the other operations to not call it again.